### PR TITLE
stepmania: fix `aarch64-linux` build by using libpng from nixpkgs

### PR DESCRIPTION
**Pull request by fx-chun**
Currently, the aarch64-linux build of Stepmania fails with libpng linkage errors:

> ld: ../extern/libpng.a(pngrtran.c.o): in function \`png_do_read_transformations':
> pngrtran.c:(.text+0x4118): undefined reference to \`png_riffle_palette_neon'
> ld: pngrtran.c:(.text+0x5188): undefined reference to \`png_do_expand_palette_rgba8_neon'
> ld: pngrtran.c:(.text+0x6268): undefined reference to \`png_do_expand_palette_rgb8_neon'
> ld: ../extern/libpng.a(pngrutil.c.o): in function \`png_read_filter_row':
> pngrutil.c:(.text+0x6734): undefined reference to \`png_init_filter_functions_neon'
> collect2: error: ld returned 1 exit status

Stepmania vendors many dependencies in source form in their code repository; it appears that the libpng build configured there seems to be either misconfigured or isn't building correctly under NixOS. I played around with a couple of NEON-related compile flags but couldn't get anything to stick.

A workaround that works seems to be building against the Nixpkgs libpng instead. It builds, launches, and (so far) plays correctly on my NixOS-based M1 Air and PC.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/stepmania/default.nix [hunk 1]
```diff
@@ -11,6 +11,7 @@
 , gtk2
 , libmad
 , libogg
+, libpng
```
[the remark this answers is not in the document]
Comment (fx-chun) [reply pkgs/games/stepmania/default.nix:14]: Huh... I wonder where it's pulling libpng from then? Maybe from the FFMPEG closure?

I added it to `buildInputs` to be explicit about it.

[the remark this answers is not in the document]
Comment (eclairevoyant) [reply pkgs/games/stepmania/default.nix:14]: I'm not sure how it's getting pulled in. I know freetype has it under `propagatedBuildInputs`, but even after removing it from there, it's still in the buildtime closure.

**Review by h7x4 — APPROVED** [review]
The diff looks good! I'm not sure what's up with the implicit libpng import though. While it would be nice to find the origin, I don't think it matters too much. Thank you for fixing :)

File: pkgs/games/stepmania/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (h7x4) [reply pkgs/games/stepmania/default.nix:14]: It seems to be through ffmpeg

```
$ nix why-depends .#stepmania .#libpng.out
/nix/store/hdzqdvifvw6v94f2y5v9l03wmmai01bg-stepmania-5.1.0-b2-unstable-2022-11-14
└───/nix/store/fj85yvx8zmakhf63xj0jkrnsih057514-ffmpeg-6.1.1-lib
    └───/nix/store/jlf6951b8ir7zisvypy26xn43jdsv9vk-fontconfig-2.15.0-lib
        └───/nix/store/h6v38d224rxbwbfj95wi3cq3gnmb9fr5-libpng-apng-1.6.43
```

**Pull request merged**